environment.yml

```diff
@@ -34,7 +34,7 @@ dependencies:
       - jupyter-client==6.1.5
       - lightning-bolts==0.4.0
       - matplotlib==3.3.0
-      - mlflow==1.17.0
+      - mlflow==1.23.1
```

Comment:
why is this needed?

Reply:
Component Governance had a high-severity alert for `mlflow`.

Reply:
OK - I think these are the changes we need to keep under control when we finally decide to decouple the histo env. It can easily drag a lot of updates in other libraries and this generates environment instability